File: packages/facebook-sign-in/ios/Plugin/FacebookSignInHelper.swift
import Foundation
import Capacitor
import FBSDKCoreKit

public class FacebookSignInHelper {
    public static func createAccessTokenJSObject(_ accessToken: AccessToken) -> JSObject {
        var result = JSObject()
        result["expiresAt"] = Int(accessToken.expirationDate.timeIntervalSince1970 * 1000)
        result["permissions"] = accessToken.permissions.map { $0.name }
        result["token"] = accessToken.tokenString
        result["userId"] = accessToken.userID
        return result
    }
}

File: packages/facebook-sign-in/android/src/main/java/io/capawesome/capacitorjs/plugins/facebooksignin/FacebookSignInHelper.java
package io.capawesome.capacitorjs.plugins.facebooksignin;

import androidx.annotation.NonNull;
import com.facebook.AccessToken;
import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;

public class FacebookSignInHelper {

    @NonNull
    public static JSObject createAccessTokenJSObject(@NonNull AccessToken accessToken) {
        JSArray permissionsResult = new JSArray();
        for (String permission : accessToken.getPermissions()) {
            if (permission != null) {
                permissionsResult.put(permission);
            }
        }
        JSObject result = new JSObject();
        result.put("expiresAt", accessToken.getExpires().getTime());
        result.put("permissions", permissionsResult);
        result.put("token", accessToken.getToken());
        result.put("userId", accessToken.getUserId());
        return result;
    }
}
